### PR TITLE
docs: restructure documentation and update formatting

### PR DESCRIPTION
**Pull request by akashsinghal**
This PR updates some of the layout of the CRD and reference docs for Ratify. It also updates some of the existing docs to be markdown formatted correctly.

### CODE REVIEW

File: docs/plugins/Store/oras.md [hunk 1]
```diff
@@ -1,10 +1,97 @@
...
+| -------------- | -------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ | --------------------------- |
+| cosignEnabled  | no       | This must be `true` if cosign verifier is enabled. Read more about cosign verifier [here](https://github.com/deislabs/ratify/blob/main/plugins/verifier/cosign/README.md).                                         | `false`                     |
+| authProvider   | no       | This is only required if pulling from a private repository. For all supported auth mode, please review [oras-auth-provider](https://github.com/deislabs/ratify/blob/main/docs/reference/oras-auth-provider.md) doc | dockerAuth                  |
+| cacheEnabled   | no       | Oras cache, cache for all referrers for a subject. Note: global cache must be enabled first                                                                                                                        | `false`                     |
```
Comment (susanshi) [inline docs/plugins/Store/oras.md:49]: curious why defaut is false

File: docs/reference/Custom Resources/certificate-stores.md [hunk 2]
```diff
@@ -1,4 +1,6 @@
-A `Certificate Store` resource defines an array of public certificates to fetch from a provider. 
+# Certificate Store (Deprecated)
```
Comment (susanshi) [inline docs/reference/Custom Resources/certificate-stores.md:1]: just a note to add a link to KMP doc once its available.

File: docs/reference/Custom Resources/certificate-stores.md [hunk 3]
```diff
@@ -46,27 +49,30 @@ status:
...
 ```
 
-| Name        | Required | Description | Default Value |
-| ----------- | -------- | ----------- | ------------- | 
-| vaultURI      | yes    |      URI of the azure key vault        |   ""            |
-| certificateName      | yes    |    the name of the key vault object   |       ""        |
-| certificateVersion   | no     |   provider will fetch latest version if empty   |     ""    |
-| tenantID   | yes     |   tenantID of the workload identity that have read access to this key vault   |     ""    |
-| clientID   | yes     |   clientID of the workload identity that have read access to this key vault   |     ""    |
+| Name               | Required | Description                                                               | Default Value |
```
Comment (susanshi) [inline docs/reference/Custom Resources/certificate-stores.md:52]: this is nice. Curious, does vscode extension auto format this?

Comment (akashsinghal) [reply docs/reference/Custom Resources/certificate-stores.md:52]: yes. I use this extension: https://marketplace.visualstudio.com/items?itemName=TakumiI.markdowntable

File: docs/reference/Custom Resources/verifiers.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# Verifier
+
```
Comment (susanshi) [inline docs/reference/Custom Resources/verifiers.md:2]: question about the path.  Should the path at lower case to be consistent with external plugins dir. If yes, we do need category.json?

File: docs/plugins/Store/oras.md (hunk 1 above)
Comment (akashsinghal) [reply docs/plugins/Store/oras.md:49]: good catch. these are not up to date. let me fix this

**Review by susanshi — APPROVED** [review]
Thanks for making docs more organized! left some questions, but can discuss/address after this merge.

File: docs/plugins/Store/oras.md (hunk 1 above)
Comment (akashsinghal) [reply docs/plugins/Store/oras.md:49]: > good catch. these are not up to date. let me fix this

Actually scratch that. It is correct in it's current form. All unspecified fields (ex: user doesn't add `cacheEnabled` in the config) will default to `false` value. This is due to behavior of `json.Unmarshal`. To set custom defaults per config, we wuold need to introduce that in ORAS store implementation. This is also why some projects never use the raw `bool` type directly. They use a pointer `*bool`. This allows for three states (nil, true, false) where the unspecified has it's own `nil` state.

File: docs/reference/Custom Resources/verifiers.md (hunk 4 above)
Comment (akashsinghal) [reply docs/reference/Custom Resources/verifiers.md]: I've updated the file paths to now be consistently lower case. For upper-case sub folders, I've added corresponding `_category_.json`

**Review by susanshi — APPROVED** [review]

**Pull request merged**